common/logger.py

```diff
@@ -1,96 +1,124 @@
+import structlog
 
+configured = False # set to True once setup_logging is called
 
-class _StructuredMessage:
-    def __init__(self, message, **kwargs):
-        self.message = message
-        self.time = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
-        self.kwargs = kwargs
+timestamper = structlog.processors.TimeStamper(fmt="%d/%m/%Y %H:%M:%S")
+pre_chain = [
+    timestamper,
+]
 
-    def to_dict(self):
-        return {**self.kwargs, "message": self.message, "time": self.time}
 
-
-class Logger:
+# Stripped down version of structlog.dev.ConsoleRenderer, adding the "component" instead of the level.
+class CustomLogRenderer:
     """
-    The :class:`Logger` is in charge of logging events into the log file.
-
-    Args:
-        log_name_prefix (:obj:`str`): the prefix of the logger where the data will be stored in (server, client, ...).
-        actor (:obj:`str`): the system actor that is logging the event (e.g. ``Watcher``, ``Cryptographer``, ...).
+    Render ``event_dict``. It renders the timestamp, followed by the component within "[]" (unless it's None),
+    followed by the event, then any remaining item in ``event_dict`` in the key=value format.
     """
 
-    def __init__(self, log_name_prefix, actor=None):
-        self.actor = actor
-        self.f_logger = logging.getLogger("{}_file_log".format(log_name_prefix))
-        self.c_logger = logging.getLogger("{}_console_log".format(log_name_prefix))
+    def _repr(self, val):
+        """Returns the representation of *val* if it's not a ``str``."""
+        return val if isinstance(val, str) else repr(val)
 
-    def _add_prefix(self, msg):
-        return msg if self.actor is None else "[{}]: {}".format(self.actor, msg)
+    def __call__(self, _, __, event_dict):
+        """Returns ``event_dict`` rendered as a string."""
+        sio = StringIO()
 
-    def _create_console_message(self, msg, **kwargs):
-        s_message = _StructuredMessage(self._add_prefix(msg), **kwargs).to_dict()
-        message = "{} {}".format(s_message["time"], s_message["message"])
+        ts = event_dict.pop("timestamp", None)
+        if ts:
+            sio.write(str(ts) + " ")
 
-        # s_message will always have at least two items (message and time).
-        if len(s_message) > 2:
-            params = "".join("{}={}, ".format(k, v) for k, v in s_message.items() if k not in ["message", "time"])
+        component = event_dict.pop("component", None)
+        if component:
+            sio.write("[" + component + "] ")
 
-            # Remove the extra 2 characters (space and comma) and add all data to the final message.
-            message += " ({})".format(params[:-2])
+        event = self._repr(event_dict.pop("event"))
 
-        return message
+        sio.write(event)
 
-    @staticmethod
-    def _create_file_message(msg, **kwargs):
-        return json.dumps(_StructuredMessage(msg, **kwargs).to_dict())
+        # Represent all the key=value elements still in event_dict
```

Comment:
We can surround this by parenthesis so it is easier to parse in case we ever need to:

ts [component] msg (key1=value1, ...)

Reply:
Good idea, done in e2601494837b42927ed1b1a4afa933c6a5336f0d.